### ISSUE TITLE
ParallelTestExecution ignores some fatal exceptions

### ISSUE THREAD

**Issue by AlterEgo7** [scalatest/scalatest#1870]
When a test suite extends `ParallelTestExecution` then runner completely ignores a test if a set of fatal exceptions are thrown.

Consider the following test:

```scala
import org.scalatest.ParallelTestExecution
import org.scalatest.funspec.AnyFunSpec
import org.scalatest.matchers.should.Matchers

class LinkageErrorTest extends AnyFunSpec with Matchers with ParallelTestExecution {

  it("should blow up") {
    throw new OutOfMemoryError("test")
    throw new LinkageError("test")
    throw new ThreadDeath
  }
}
```
produces the following output:
```
LinkageErrorTest:
Execution took 0ms
No test suite was run
```

If the mixin of `ParallelTestExecution` is removed then the output is correct:
```
LinkageErrorTest:
product.canon.services.v1.secretService.LinkageErrorTest *** ABORTED ***
  java.lang.OutOfMemoryError: test
  at product.canon.services.v1.secretService.LinkageErrorTest.$anonfun$new$1(LinkageErrorTest.scala:10)
  at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
  at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
  at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
  at org.scalatest.Transformer.apply(Transformer.scala:22)
  at org.scalatest.Transformer.apply(Transformer.scala:20)
  at org.scalatest.funspec.AnyFunSpecLike$$anon$1.apply(AnyFunSpecLike.scala:456)
  at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
  at org.scalatest.TestSuite.withFixture$(TestSuite.scala:195)
  at org.scalatest.funspec.AnyFunSpec.withFixture(AnyFunSpec.scala:1631)
  ...
[E] Uncaught exception when running product.canon.services.v1.secretService.LinkageErrorTest: java.lang.OutOfMemoryError: test
[E] Test suite aborted
Execution took 0ms
1 tests, 1 errors
```

**Note:**  If the tests are run through Intellij IDEA using `ParallelTestExecution`, the error is showed.

**Comment by cheeseng** [comment]
@AlterEgo7  I just released a 3.2.2-M2, do you mind to try to see if it fixes your problem?

Cheers.

**Comment by AlterEgo7** [comment]
Hi @cheeseng , I can confirm that 3.2.2-M2 fixes the issue.

Feel free to edit / close the issue.
